src/datachain/utils.py

```diff
@@ -251,6 +251,21 @@ def batched(iterable: Iterable[_T_co], n: int) -> Iterator[tuple[_T_co, ...]]:
         yield batch
 
 
+def batched_it(iterable: Iterable[_T_co], n: int) -> Iterator[Iterator[_T_co]]:
+    """Batch data into iterators of length n. The last batch may be shorter."""
+    # batched('ABCDEFG', 3) --> ABC DEF G
+    if n < 1:
+        raise ValueError("Batch size must be at least one")
+    it = iter(iterable)
+    while True:
+        chunk_it = islice(it, n)
+        try:
+            first_el = next(chunk_it)
+        except StopIteration:
+            return
```

Comment:
could you remind me - does it mean it can return None? should it be also throwing StopIteration (to behave like an iterator I guess)

Reply:
Nice catch, let me check 🙏

Reply:
When a generator returns, `StopIteration` is raised in the caller. Raising `StopIteration` inside the generator would be wrong, see https://peps.python.org/pep-0479/